components/brave_ads/browser/background_helper.cc

```diff
@@ -33,11 +34,15 @@ void BackgroundHelper::RemoveObserver(Observer* observer) {
   observers_.RemoveObserver(observer);
 }
 
-#if !defined(OS_MACOSX) && !defined(OS_WIN) && !defined(OS_LINUX)
 BackgroundHelper* BackgroundHelper::GetInstance() {
+#if defined(OS_ANDROID)
+  return  BackgroundHelperAndroid::GetInstance();
```

Comment:
We should not need this code as the change below `BackgroundHelper* BackgroundHelper::GetInstance()` will make sure that the Android implementation is called

Reply:
The above approach has been used for LocaleHelper and I followed it in case of BackgroundHelper.  
Changing it for LocaleHelper too.